src/main/java/com/mongodb/jdbc/MongoDriver.java

```diff
@@ -221,25 +239,26 @@ private MongoConnection createConnection(ConnectionString cs, Properties info)
         switch (dialect.toLowerCase()) {
             case MYSQL_DIALECT:
                 return new MySQLConnection(
-                        cs,
-                        info.getProperty(DATABASE),
-                        info.getProperty(CONVERSION_MODE),
-                        logLevel,
-                        logDir);
+                        cs, database, info.getProperty(CONVERSION_MODE), logLevel, logDir);
             case MONGOSQL_DIALECT:
                 if (info.containsKey(CONVERSION_MODE)) {
-                    throw new SQLClientInfoException(
-                            String.format(
-                                    "must not set '%s' if '%s' is '%s'",
-                                    CONVERSION_MODE, DIALECT, MONGOSQL_DIALECT),
-                            Collections.singletonMap(
-                                    CONVERSION_MODE, ClientInfoStatus.REASON_VALUE_INVALID));
+                    throw new SQLException(
+                            "Must not set"
+                                    + CONVERSION_MODE
+                                    + " if "
+                                    + DIALECT
+                                    + " is "
+                                    + MONGOSQL_DIALECT);
                 }
-                return new MongoSQLConnection(cs, info.getProperty(DATABASE), logLevel, logDir);
+                return new MongoSQLConnection(cs, database, logLevel, logDir);
             default:
-                throw new SQLClientInfoException(
-                        String.format("invalid dialect '%s'", dialect),
-                        Collections.singletonMap(DIALECT, ClientInfoStatus.REASON_VALUE_INVALID));
+                throw new SQLException(
```

Comment:
Good catch, it looks like `SQLClientInfoException` is mainly thrown in the `setClientInfo()` method.